### DISCUSSION TITLE
Will there be performance issues when using the Image control to display videos

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#17740] [Q&A] [answered]
I want to use Avalonia as a cross platform video playback software. The solution I have come up with now is to obtain the decoded raw data for each frame (Rbga8888), and then create a bitmap to assign a value to the source of the image tag to achieve the playback effect of the video.
The demo display was successful, but in actual scenarios, there may be multiple displays with multiple camera screens on each display. Will there be any drawing issues using this method where Avalonia draws it herself.

**Comment by thevortexcloud** [comment, marked as answer]
If you need high performance, make sure you use `WriteableBitmap`.

**Reply by ghost** [reply]
Thanks,I'll try it

**Reply by thevortexcloud** [reply]
Do keep in mind that you probably want to keep as much as you can off the CPU. But that requires things like the OpenGL control.

You can also use VLC Sharp, which has official Avalonia support if you don't mind dealing with the problems of embedding the VLC Window inside Avalonia.

https://code.videolan.org/videolan/LibVLCSharp/-/tree/master/samples/LibVLCSharp.Avalonia.Sample

**Reply by ghost** [reply]
![微信截图_20241210164139](https://github.com/user-attachments/assets/e5cf959c-e5e3-44b1-841d-ae9f40c951cc)
Look at this.
I use the WriteableBitmap.
I updated the image data in the memory, but the image tag only displayed my first image. When I updated the image data in the memory later, it did not refresh the drawing. Is there any way to make the image display refresh the drawing?

**Reply by maxkatz6** [reply]
`image.InvalidateVisual()` needs to be called, when you update bitmap, IIRC

**Reply by ghost** [reply]
> `image.InvalidateVisual()` needs to be called, when you update bitmap, IIRC

Thank you, I succeeded
